source/core_mqtt.c

```diff
@@ -1024,7 +1025,11 @@ static MQTTStatus_t handleKeepAlive( MQTTContext_t * pContext )
         }
         else
         {
-            if( calculateElapsedTime( now, pContext->lastPacketTime ) > keepAliveMs )
+            if( calculateElapsedTime( now, pContext->lastPacketTxTime ) >= keepAliveMs )
```

Comment:
Why should we check for both RX and TX timestamp expires ? IMO we should only check for RX timestamp expiry here as that would cover both cases.